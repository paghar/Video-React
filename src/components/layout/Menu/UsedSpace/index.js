//@flow
import React from 'react';
import { LinkDiv, UsedSpaceText } from './index.style';
import { Display3 } from 'theme/fonts';
import { Generals } from 'theme/colors';

type Props = {
  totalSpace: string,
  usedSpace: string,
  className?: string,
};

function usedSpace({ usedSpace, totalSpace, className }: Props) {
  return (
    <div className={className}>
      <UsedSpaceText font={Display3} color={Generals.viii} hoverable>
        {usedSpace} GB of {totalSpace} GB used
      </UsedSpaceText>

      <LinkDiv font={Display3} color={Generals.vi} hoverable>
        Upgrade Now
      </LinkDiv>
    </div>
  );
}

export default usedSpace;
