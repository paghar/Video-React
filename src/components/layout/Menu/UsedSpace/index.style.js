//@flow
import styled from 'styled-components';
import { Generals } from 'theme/colors';
import Typography from 'components/shared/Typography';

export const UsedSpaceText = styled(Typography)`
  padding-bottom: 8px;
  display: block;
`;

export const LinkDiv = styled(Typography)`
  text-decoration-line: underline;
  text-decoration-style: solid;
  color: ${Generals.vi};
`;
