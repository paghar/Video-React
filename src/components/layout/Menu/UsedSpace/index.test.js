//@flow
import React from 'react';
import renderer from 'react-test-renderer';
import UsedSpace from '.';

describe('<UsedSpace />', () => {
  it('when Menu in open mode', () => {
    const tree = renderer
      .create(<UsedSpace usedSpace="80" totalSpace="250"></UsedSpace>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('<UsedSpace />', () => {
  it('when Menu in close mode', () => {
    const tree = renderer
      .create(<UsedSpace usedSpace="80" totalSpace="250"></UsedSpace>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
