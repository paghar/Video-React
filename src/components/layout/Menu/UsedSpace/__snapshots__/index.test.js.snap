// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UsedSpace /> when Menu in close mode 1`] = `
.c0 {
  pointer-events: none;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c1 {
  padding-bottom: 8px;
  display: block;
}

.c2 {
  -webkit-text-decoration-line: underline;
  text-decoration-line: underline;
  -webkit-text-decoration-style: solid;
  text-decoration-style: solid;
}

<div>
  <span
    className="c0 c1"
  >
    80
     GB of 
    250
     GB used
  </span>
  <span
    className="c0 c2"
  >
    Upgrade Now
  </span>
</div>
`;

exports[`<UsedSpace /> when Menu in open mode 1`] = `
.c0 {
  pointer-events: none;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c1 {
  padding-bottom: 8px;
  display: block;
}

.c2 {
  -webkit-text-decoration-line: underline;
  text-decoration-line: underline;
  -webkit-text-decoration-style: solid;
  text-decoration-style: solid;
}

<div>
  <span
    className="c0 c1"
  >
    80
     GB of 
    250
     GB used
  </span>
  <span
    className="c0 c2"
  >
    Upgrade Now
  </span>
</div>
`;
